src/libutil/error.cc

```diff
@@ -334,7 +357,15 @@ std::ostream & showErrorInfo(std::ostream & out, const ErrorInfo & einfo, bool s
 
     bool frameOnly = false;
     if (!einfo.traces.empty()) {
+        std::set<Trace> tracesSeen;
+        size_t skippedTraces = 0;
         size_t count = 0;
+        const auto printSkippedTracesMaybe = [&] {
+            if (skippedTraces > 0) {
+                oss << "\n" << ANSI_WARNING "(" << skippedTraces << " duplicate traces omitted)" ANSI_NORMAL << "\n";
```

Comment:
Okay, I took another shot at this. Instead of storing an integer of `skippedTraces`, I store a `std::vector` of them, and then I can just check if there's only a couple skipped and print them manually (to avoid "1 duplicate frame omitted" messages).

I'm also clearing the set of seen traces each time we print "duplicate frames omitted", so that we don't accidentally hide interesting parts of the trace. Of course, this could break down in more complex recursion scenarios...

Reply:
Yeah, I am contemplating the cases:
- Bunch of repeated stuff, finally a different stack frame (this is improbable tho, why would it be 10000 specifically)
- Mutual recursion goes and infinitely recurses
  - If the pattern is *just* a repeated run of A, B, C, A, B, C, in unicorns-and-rainbows land it would recognize the repeated run and truncate just the run? But it would also be ok if we don't notice that; this change is a big improvement over the status quo already.
  - If the pattern is more than that, idk, feels like the upper parts of the stack trace are going to be useless regardless, and users would be better served by the debugger and a breakpoint.
- One function infinitely recurses
  - Here we definitely want to chop the stack trace since it's useless.

We might also want a total limit of how much stack trace we print and `--srsly-show-me-all-of-it`? Kinda like RUST_BACKTRACE=1 vs RUST_BACKTRACE=all.